src/components/editor/FileSelector.tsx

```diff
@@ -1,51 +1,97 @@
 import { EditorStoreState, useZustandStore } from 'components/editor/Store';
 import { DraftSpecQuery } from 'hooks/useDraftSpecs';
+import { useEffect, useState } from 'react';
 import { FormattedMessage } from 'react-intl';
 
-function EditorFileSelector() {
-    const currentCatalog = useZustandStore<
-        EditorStoreState<LiveSpecQuery | DraftSpecQuery>,
-        EditorStoreState<LiveSpecQuery | DraftSpecQuery>['currentCatalog']
-    >((state) => state.currentCatalog);
+const initialState = {
+    columns: {
+        columnVisibilityModel: {
+            spec_type: false,
+        },
+    },
+};
+
+const getRowId = (spec: any) => {
+    let newSelectionModel;
+
+    if (spec.id) {
+        newSelectionModel = spec.id;
+    } else if (spec.draft_id) {
+        newSelectionModel = spec.draft_id;
+    }
+
+    return `${newSelectionModel}-${spec.catalog_name}`;
+};
+
+const columns: GridColDef[] = [
+    {
+        field: 'catalog_name',
+        headerName: 'Files',
+        flex: 1,
+        renderCell: (params: GridRenderCellParams<Date>) => (
+            <ListItemText
+                primary={params.row.catalog_name}
+                secondary={params.row.spec_type}
+            />
+        ),
+    },
+    {
+        field: 'spec_type',
+        headerName: 'Type',
+    },
+];
 
+function EditorFileSelector() {
     const setCurrentCatalog = useZustandStore<
         EditorStoreState<LiveSpecQuery | DraftSpecQuery>,
         EditorStoreState<LiveSpecQuery | DraftSpecQuery>['setCurrentCatalog']
     >((state) => state.setCurrentCatalog);
 
-    const id = useZustandStore<
-        EditorStoreState<LiveSpecQuery | DraftSpecQuery>,
-        EditorStoreState<LiveSpecQuery | DraftSpecQuery>['id']
-    >((state) => state.id);
-
     const specs = useZustandStore<
         EditorStoreState<LiveSpecQuery | DraftSpecQuery>,
         EditorStoreState<LiveSpecQuery | DraftSpecQuery>['specs']
     >((state) => state.specs);
 
-    return (
-        <List dense disablePadding>
-            {!id ? null : specs && specs.length > 0 ? (
-                specs.map((tag: any, index: number) => (
-                    <ListItemButton
-                        key={`FileSelector-${tag.catalog_name}`}
-                        dense
-                        selected={index === currentCatalog}
-                        onClick={() => setCurrentCatalog(index)}
-                    >
-                        <ListItemText
-                            primary={tag.catalog_name}
-                            secondary={tag.spec_type}
-                        />
-                    </ListItemButton>
-                ))
-            ) : (
-                <FormattedMessage id="common.loading" />
-            )}
-        </List>
+    const [selectionModel, setSelectionModel] = useState<GridSelectionModel>(
+        []
     );
+
+    useEffect(() => {
+        if (specs) {
+            setSelectionModel(getRowId(specs[0]) as any);
+        }
+    }, [specs]);
```

Comment:
This basically sets the first spec as the auto selected one. This does not _really_ work in all cases so some work needs to be done. Probably just need to add hacky "initialSettingDone" sort of thing so this does not happen repeatedly.